tutorials/snake/steps/step-2.md

```diff
@@ -34,16 +36,16 @@ Then to compute the index in that sequence from a set of 2D coordinates, we can
-    2. `(->)` is a block container that will executes its inner block(s) in order.
-    3. `(|)` is an alias for [`(Sub)`](https://docs.fragcolor.xyz/blocks/General/Sub/). It allows to reuse the same input in the next block.
-    4. [`(Take)`](https://docs.fragcolor.xyz/blocks/General/Take/) returns the value from a sequence at a given index (starting at `0`).
-    5. `>=` saves the output of a block into a context variable.
-    6. [`(Math.Multiply)`](https://docs.fragcolor.xyz/blocks/Math/Multiply/) multiplies its input with a given value and outputs the result.
-    7. [`(Math.Add)`](https://docs.fragcolor.xyz/blocks/Math/Add/) adds a value to its input and outputs the result.
+    1. The [`defn`](https://docs.fragcolor.xyz/functions/macros/#defn) keyword associates a function with a name. Note the `[]` after the `get-index` name. This indicates that this function has 0 parameters. We will later see functions that do have parameters.
+    2. [`(->)`](https://docs.fragcolor.xyz/functions/misc/#shard-container) is a shard container that will group and execute its inner shard(s) in order.
+    3. `(|)` is an alias for [`(Sub)`](https://docs.fragcolor.xyz/shards/General/Sub/). It allows reusing the same input across a sequence of shards.
+    4. [`(Take)`](https://docs.fragcolor.xyz/shards/General/Take/) returns the value from a sequence at a given index (starting at `0`).
+    5. `>=` is an alias for the shard [`(Set)`](https://docs.fragcolor.xyz/shards/General/Set/) which saves the output of ashard into a context variable.
+    6. [`(Math.Multiply)`](https://docs.fragcolor.xyz/shards/Math/Multiply/) multiplies its input (written to the left of theshard) with a given value (written to the right of theshard and enclosed within it’s brackets) and outputs the result.
+    7. [`(Math.Add)`](https://docs.fragcolor.xyz/shards/Math/Add/) adds a value to its input and outputs the result.
         
     ??? note
-        Because `defn` expects a single "value" after the name and the list of parameters, a `(->)` block is required to group several blocks together. As it is a common occurrence, a shortcut is also provided: `defblocks`.
+        Because defn expects a single "value" (called function return value) after the function name and the list of parameters, and our function’s logic (function body) contains multipleshards, a (->)shard is required  used required  to group these severalshards together in a single (return)shard . Since this is a common situation with `(defn)` functionshards, a convenient alternative is to use  [`(defblocks)`](https://docs.fragcolor.xyz/functions/macros/#defblocks) instead. A `(defblock)` behaves exactly like a function (including the ability to accept input parameters) but can contain multipleshards in it’s body. These multipleshards are executed in the order that they appear and `(defblocks)` return value is the output of the lastshard in its body. .
```

Comment:
Lots of missing white spaces and there is a repetition of `required` in the middle. In addition:
- `it's` -> `its`
- `(defblocks)`-> `(defshards)`

Reply:
Same as above comment.